doc/method_documentation.rdoc

```diff
@@ -181,3 +181,26 @@ of the method.
 Methods are documented using RDoc syntax.  See the
 {RDoc Markup Reference}[https://docs.ruby-lang.org/en/master/RDoc/Markup.html#class-RDoc::Markup-label-RDoc+Markup+Reference]
 for more information on formatting with RDoc syntax.
+
+=== Code Blocks
+
+A code block should be preceded and followed by a blank line.
+This is unnecessary for the HTML output, but helps in the ri output.
+
+Consider whether <tt>=> ...</tt> in successive lines should be aligned.
+Alignment may sometimes aid readability.
+
+=== Lists
+
+A list should be preceded and followed by a blank line.
+This is unnecessary for the HTML output, but helps in the ri output.
+
+=== Auto-Links
+
+Consider whether an auto-link (e.g., to +Array+)
+should be suppressed (<tt>\Array</tt>) or allowed (Array).
+
+It's usually best to suppress when local;
+that is, in the documentation for class \Array, suppress auto-links to the class itself.
+
+For an "off-class" reference, the decision to suppress or not may depend on context.
```

Comment:
Not a strong one, no. "depend on context" sounds fine to me. The base classes are so common that I think that one can also use "array", "integer", "string"...

Reply:
I think for "off-class" references they highly help for discoverability of relevant other documentation, especially for those users who are not deeply familiar with Ruby (this is less a point for the most common classes like arrays, integers, strings, where I agree with @marcandre). "depend on context" seems ok, but I would usually _not_ suppress such links.

Reply:
The usual 'previous practice' is to not cite the class at all, saying string instead of String, integer instead of Integer, hash instead of Hash, array instead of Array.  Do you think those many of hundreds (maybe thousands) should be changed to class names with generated links?

Reply:
As before, I'd like to hear much more from others.

Reply:
I'm fine removing the Auto-Links section and leaving it up to the person writing the method documentation whether to suppress the links or not.  I personally don't have strong feelings either way.  Auto-linking can be safely ignored if present, and doesn't add much value if not present.